core/src/main/scala/org/http4s/Message.scala

```diff
@@ -359,28 +477,47 @@ final case class Response[F[_]](
     attributes = attributes
   )
 
-  override def withStatus(status: Status)(implicit F: Functor[F]): Self =
+  def withStatus(status: Status): Self =
     copy(status = status)
 
-  def withBodyStream(body: EntityBody[F]): Self =
-    copy(body = body)
-
   override protected def change(
+      httpVersion: HttpVersion,
       body: EntityBody[F],
       headers: Headers,
-      attributes: AttributeMap): Self =
-    copy(body = body, headers = headers, attributes = attributes)
+      attributes: AttributeMap
+  ): Self =
+    copy(
+      httpVersion = httpVersion,
+      body = body,
+      headers = headers,
+      attributes = attributes
+    )
 
-  override def toString: String = {
-    val newHeaders = headers.redactSensitive()
-    s"""Response(status=${status.code}, headers=$newHeaders)"""
-  }
+  /** Add a Set-Cookie header for the provided [[ResponseCookie]] */
+  def addCookie(cookie: ResponseCookie): Self =
```

Comment:
Why is it `addCookie` when it's `putHeaders`?  Is there a reason we should be inconsistent here?

Reply:
It's `headers.put` underneath, but deciding on `add` instead would be fine by me.

Reply:
I guess `addHeaders` would be wrong because it replaces singleton headers.  I retract the suggestion.